source/ota_over_mqtt_demo.c

```diff
@@ -147,7 +150,26 @@
 #define APP_VERSION_MINOR                  9
 #define APP_VERSION_BUILD                  2
 
-typedef void ( * PublishCallback_t ) ( MQTTPublishInfo_t *, void * );
+/**
+ * @brief Defines the structure to keep the subscription info so that it can be accessed
+ * in the command complete callback in this demo.
+ */
+typedef struct SubscriptionInfo
+{
+    IncomingPubCallback_t pxIncomingPublishCallback;
+    const char * pcTopicFilter;
+} SubscriptionInfo_t;
```

Comment:
It looks odd that the filter length is not part of this. Also, adding a boolean flag `isSubscribe` would allow you to use the same callback for both subscribes and unsubscribes, since the only difference is the call to `addSubscription`/`removeSubscription`

Reply:
That is a good suggestion.